lib/spells/CSpellHandler.h

```diff
@@ -191,17 +185,17 @@ class DLL_LINKAGE CSpell : public spells::Spell
 
 	using BTVector = std::vector<Bonus::BonusType>;
 
-	si32 level;
+	si32 level = 0;
 
 	std::map<ESpellSchool, bool> school;
 
-	si32 power; //spell's power
+	si32 power = 0; //spell's power
 
 	std::map<TFaction, si32> probabilities; //% chance to gain for castles
 
-	bool combat; //is this spell combat (true) or adventure (false)
-	bool creatureAbility; //if true, only creatures can use this spell
-	si8 positiveness; //1 if spell is positive for influenced stacks, 0 if it is indifferent, -1 if it's negative
+	bool combat = false; //is this spell combat (true) or adventure (false)
+	bool creatureAbility = false; //if true, only creatures can use this spell
+	si8 positiveness = ESpellPositiveness::NEUTRAL; //1 if spell is positive for influenced stacks, 0 if it is indifferent, -1 if it's negative
```

Comment:
I think better to initialize all of these in constructor.
1) This creates additional "noise" if you simply want to check public interface of a class
2) This also requires providing access to enum values in header files. Meaning, additional includes in header that can't be removed, slowing down compilation, especially - rebuilds where you have to change some common header

IMO in-place initialization is nice for small C-like structs, but for classes better to use constructor init

Reply:
@IvanSavenko is constructor removed here? If this struct becomes trivial - then I think we should initialize all fields in class there. If not - we can again create default constructor with initialized fields.

Reply:
moved to constructor again